### PR TITLE
chore: bump dependencies for k8s v1.25.3

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Bump kubernetes dependencies to `v1.25.3`
- Bump cert-controller to `v0.5.0`
- Bump microsoft-authentication-library-for-go to `v0.7.0`

**Requirements**

- [ ] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [ ] no

**Notes for Reviewers**:

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -34,9 +34,9 @@ require (
...
 	gopkg.in/ini.v1 v1.62.1
 	gopkg.in/square/go-jose.v2 v2.6.0
-	k8s.io/api v0.25.0
-	k8s.io/apimachinery v0.25.0
-	k8s.io/client-go v0.25.0
+	k8s.io/api v0.25.3
+	k8s.io/apimachinery v0.25.3
+	k8s.io/client-go v0.25.3
 	k8s.io/klog/v2 v2.80.1
 	k8s.io/kubernetes v1.25.3
```
Comment (enj) [inline go.mod:41]: 😱 ??

File: go.mod [hunk 2]
```diff
@@ -139,31 +139,31 @@ require (
 )
 
 replace (
```
Comment (enj) [inline go.mod:141]: Why do we have all these kube deps?  And why are they all pinned ?

Comment (aramase) [reply go.mod:141]: `k8s.io/api` and `k8s.io/apimachinery` are used in the webhook. `k8s.io/client-go` is used in e2e tests.

Comment (aramase) [reply go.mod:141]: Also, we use `k8s.io/kubernetes/test/e2e/framework` in e2e which requires these deps to be pinned.

**Review by enj — APPROVED** [review]

**Pull request merged**